client/v3/naming/endpoints/endpoints.go

```diff
@@ -32,7 +34,14 @@ type Endpoint struct {
 	// Metadata is the information associated with Addr, which may be used
 	// to make load balancing decision.
 	// Since etcd 3.1
+	//
+	// Deprecated: use Attributes instead
 	Metadata any
+
+	// Attributes contains arbitrary data about this address intended for
+	// consumption by the SubConn.
+	// Since etcd 3.5
+	Attributes *attributes.Attributes
```

Comment:
We can assign this map to attributes.Attributes using the utilities `New` and `WithValue`

https://github.com/grpc/grpc-go/blob/4cedec40eb2ccfbe3f56bb15e894903111ada2d2/attributes/attributes.go#L44-L62 

```suggestion
	Attributes map[string]any
```

Reply:
Unfortunately the the grpc attributes package is also `Experimental`

Reply:
This approach will solve the JSON problem for sure. I will adjust it now. Meanwhile, you also mentioned the grpc attributes package is also `Experimental`. I mean we've already decided to take this direction so, things like this should be expected. Except you have something else in mind, of course. Are you having second thoughts? 

cc: @ahrtr

Reply:
> Are you having second thoughts?

The alternative is to support generic type (a comparable type), so that the key of the map doesn't have to be a string. But let's keep it simple for now, as we are playing with grpc-go's experimental feature/packages.

Reply:
+1 you should not use any experimental APIs in your own APIs, unless you want to declare them as experimental and your users are aware that you reserve the right to break them in the future.

You also need that disclaimer on any package that internally uses experimental APIs.  But if you refrain from exposing it in your own APIs, then there should be some path forward that doesn't require changes to the user's code, too.